components/BarcodeScanner.vue

```diff
@@ -0,0 +1,334 @@
+      v-show="!loadingCamera && !scanning"
+      id="guide"
+      style="position: absolute; top:0; height: 100%; width: 100%" />
+    <!-- Tip Text -->
+    <div
+      v-if="!loadingCamera && !scanning"
+      class="text-white"
+      style="position: absolute;"
+      :style="`bottom: calc(${regionTop}% - 30px)`">
+      {{tipText}}
+    </div>
+    <!-- Close -->
+    <q-btn
+      v-if="!loadingCamera && !scanning"
+      flat
+      fab
+      :ripple="false"
+      size="16px"
+      color="white"
+      icon="fas fa-times"
+      class="q-ma-sm close-btn"
+      @click.native="close()" />
+    <!-- Buttons -->
+    <Buttons
+      :camera-list="cameraList"
+      @updateCamera="updateCamera($event)"
+      @upload="upload($event)" />
+  </div>
+</template>
+
+<script>
+/*!
+  * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+  */
+import DBR from '../helpers/dbr.js';
+import driverLicenseFields from '../helpers/driverLicenseFields.js';
+import Buttons from './Buttons.vue';
+import Spinner from './Spinner.vue';
+
+DBR.BarcodeReader.license = '100902848';
+
+export default {
+  name: 'BarcodeScanner',
+  components: {
+    Buttons,
+    Spinner
+  },
+  props: {
+    tipText: {
+      type: String,
+      default: 'Scan barcode on the back of license.'
+    },
+    guideColor: {
+      type: String,
+      default: 'rgb(254, 142, 20)'
+    }
+  },
+  data() {
+    return {
+      loadingCamera: true,
+      scanning: false,
+      isDestroyed: false,
+      scanner: null,
+      reader: null,
+      dlInfo: null,
+      clientHeight: null,
+      clientWidth: null,
+      regionScale: 0.4,
+      cameraList: [],
+      currentCamera: null,
+      maskCanvas: null,
+      maskCanvasContext: null,
+    };
+  },
+  computed: {
+    regionMaskEdgeLength() {
+      const regionMaskEdgeLength =
+        this.regionScale * Math.min(this.clientWidth, this.clientHeight);
+      return Math.floor(regionMaskEdgeLength);
+    },
+    regionLeft() {
+      const left =
+        (this.clientWidth - this.regionMaskEdgeLength) / 2 / this.clientWidth;
+      if(this.clientWidth > this.clientHeight) {
+        return Math.round(left * 100) - 25;
+      } else {
+        return Math.round(left * 100) - 20;
+      }
+    },
+    regionTop() {
+      const top =
+        (this.clientHeight - this.regionMaskEdgeLength) / 2 / this.clientHeight;
+      return Math.round(top * 100) - 5;
+    },
+    region() {
+      return {
+        regionLeft: this.regionLeft,
+        regionRight: 100 - this.regionLeft,
+        regionTop: this.regionTop,
+        regionBottom: 100 - this.regionTop,
+        regionMeasuredByPercentage: 1,
+      };
+    },
+  },
+  async mounted() {
+    this.clientHeight = document.body.clientHeight;
+    this.clientWidth = document.body.clientWidth;
+    this.maskCanvas = document.getElementById('guide');
+    this.maskCanvasContext = this.maskCanvas.getContext('2d');
+    this.cvsDrawArea();
+    window.onresize = () => {
```

Comment:
We probably want to debounce this. I'll add an issue. I imagine if you did this on desktop and dragged the corner of the window you may cripple your CPU.

https://github.com/digitalbazaar/bedrock-vue-pdf417/issues/7